
```{r}
library(qtl)
library(tidyverse)
```

```{r}
theme_set(theme_bw(base_size=14))
theme_update(panel.grid.major = element_blank(),
  panel.grid.minor = element_blank(), 
  plot.title = element_text(),
  panel.margin = unit(0, "lines"),
  strip.background = element_blank(),
  panel.border = element_blank(),
  legend.position = "none", 
  strip.text.x = element_text(margin = margin(0,0,0,0, "cm")))

chrom_names <- c(
  'NC_058337.1'="1",
  'NC_058338.1'="2",
  'NC_058339.1'="3",
  'NC_058340.1'="4",
  'NC_058341.1'="5",
  'NC_058342.1'="6")

```

Deleted header from file, here was the information removed
Stacks v2.53; R/QTL; March 31, 2022
Exported: populations.20220331
Map Type: F2
Num Loci: 121106
Num Samples: 93

# Stacks v2.53; R/QTL; April 11, 2022
# Exported: populations.20220411
# Map Type: F2
# Num Loci: 11091
# Num Samples: 93

then convert from tsv to csv 
sed 's/\t/,/g' song_family_filter_parfixed_f1het_hwe.p.F2.rqtl.tsv > song_family_gen.csv

## Read in the data files

```{r}
cross <- read.cross("csvs", 
           dir="~/Documents/Song_Genes_Paper/Rqtl", 
           genfile = "song_family_gen.csv", 
           phefile = "pxa_phen.csv",
           na.strings = "-",
           genotypes=c("a","h","b"),
           alleles=c("a","b"))
cross <- rescalemap(cross, 1e-6)
```



## Calculate cross summary statistics

```{r}
nind(cross)
nphe(cross)
nchr(cross)
totmar(cross)
```


## Plot missingness data for loci and individuals

Some individuals and markers have very high amounts of missing data which could indicate problems with DNA quality for those individuals or loci construction for that marker. Those with the highest amounts of missing data will be pruned.

```{r}
plot(ntyped(cross), ylab="# typed markers", main="# genotypes by individual", pch = 1, cex = 1)
plot(ntyped(cross, "mar"), ylab="# typed individuals", main="# genotypes by marker", pch = 1, cex = 0.2, col=alpha(rgb(0,0,0), 0.1))
```


### Remove individuals not genotyped

```{r}
cross <- subset(cross, ind = (ntyped(cross) > 1))
```

# Plot genotype frequencies by individual


```{r}
g <- pull.geno(cross)
gfreq <- apply(g, 1, function(a) table(factor(a, levels=1:3)))
gfreq <- t(t(gfreq) / colSums(gfreq))
par(mfrow=c(1,3), las=1)
for(i in 1:3)
plot(gfreq[i,], ylab="Genotype frequency", main=c("AA", "AB", "BB")[i],
ylim=c(0,1))
```



# QTL mapping

```{r}
cross <- calc.genoprob(cross, step=1, error.prob=0.01)
out.cross <- scanone(cross, pheno.col=2)
```


```{r}
out.prem <- scanone(cross, n.perm=1000, pheno.col=2)
```


```{r}
summary(out.prem,alpha=0.01)
```

```{r}

#jpeg("~/Documents/Song_Genes_Paper/Rqtl/scanone_plot_8.16.22.jpg", width = 12, height = 6, units = "cm", res = 700)
ggplot(out.cross, aes(x = pos, y = lod, color = chr)) +
  geom_point(cex = 1) +
  geom_hline(yintercept = 0, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 1, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 2, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 3, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 4, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 5, color = "grey", cex = 0.1, alpha = 0.5) +
  geom_hline(yintercept = 6, color = "grey", cex = 0.1, alpha = 0.5) +
  labs(y = "LOD", x =  "Chromosome", title = "", tag = "A") +
  scale_color_manual(values = rep(c("grey0", "grey44"), times = 3)) +
  scale_y_continuous( breaks = c( 2, 4,  6),
  labels = c( "2.0", "4.0", "6.0")) +
  geom_hline(yintercept = 4.08, color = "red", linetype = "dashed") +
  facet_wrap(~chr, ncol = 6, scale = "free_x", strip.position="bottom", labeller=as_labeller(chrom_names))
#dev.off()
```


ID marker closest to the QTL peak

```{r}
max(out.cross,chr = "NC_058337.1")
max(out.cross,chr = "NC_058338.1")
mar1 <- find.marker(cross, chr="NC_058337.1", pos=86.7)
mar2 <- find.marker(cross, chr="NC_058338.1", pos=59.2)
pxg1 <- plotPXG(cross, marker = mar1, pheno.col = 2)
pxg2 <- plotPXG(cross, marker = mar2, pheno.col = 2)

```
```{r}
#jpeg("~/Documents/Song_Genes_Paper/Rqtl/genoxpheno_chr1_plot",  width = 12, height = 8, units = "cm", res = 700)

pxg1 %>%
  ggplot(aes(y=pheno, x=as.factor(`1867`), col = as.factor(`1867`), fill = as.factor(`1867`))) +
  geom_boxplot(alpha = 0.4, outlier.shape = NA, notch = TRUE) +
  geom_point(position=position_jitterdodge(jitter.width = 0.6), cex = 3) +
  labs(y= "Volley period (seconds)", x = "Genotype", tag = "A") +
  scale_x_discrete(breaks=1:3, labels=c("AA", "AB", "BB")) +
  scale_fill_manual(values=c("black", "black", "black")) +
  scale_color_manual(values=c("black", "black", "black")) 

#dev.off()
```



```{r}
#jpeg("~/Documents/Song_Genes_Paper/Rqtl/genoxpheno_chr2_plot",  width = 12, height = 8, units = "cm", res = 700)

pxg2 %>%
  ggplot(aes(y=pheno, x=as.factor(`4353`), col = as.factor(`4353`), fill = as.factor(`4353`))) +
  geom_boxplot(alpha = 0.4, outlier.shape = NA, notch = TRUE) +
  geom_point(position=position_jitterdodge(jitter.width = 0.6), cex = 3) +
  labs(y= "Volley period (seconds)", x = "Genotype", tag = "B") +
  scale_x_discrete(breaks=1:3, labels=c("AA", "AB", "BB")) +
  scale_fill_manual(values=c("black", "black", "black")) +
  scale_color_manual(values=c("black", "black", "black")) 

#dev.off()
```

